packages/react-components/react-overflow/stories/Overflow/Pinned.stories.tsx

```diff
@@ -1,51 +1,119 @@
-  },
-
-  overflowContainer: {
-    display: 'flex',
-    flexGrow: 1,
     flexWrap: 'nowrap',
     minWidth: 0,
     ...shorthands.overflow('hidden'),
   },
-
-  farItems: {
-    dislay: 'flex',
-    ...shorthands.gap('4px'),
-    flexWrap: 'nowrap',
-    marginRight: '10px', //to allow the resize handle to be grabbed
-  },
 });
 
 export const Pinned = () => {
-  const itemIds = new Array(8).fill(0).map((_, i) => i.toString());
   const styles = useStyles();
 
+  const [selected, setSelected] = React.useState<string>('6');
+
+  const onSelect = (itemId: string) => {
+    setSelected(itemId);
+  };
+
+  const itemIds = new Array(8).fill(0).map((_, i) => i.toString());
+
   return (
-    <div className={styles.container}>
-      <Overflow>
-        <div className={styles.overflowContainer}>
+    <Overflow>
+      <div className={styles.container}>
+        {itemIds.map(i => (
+          <OverflowSelectionItem onSelectItem={onSelect} key={i} id={i} selected={selected === i} />
+        ))}
+        <OverflowMenu itemIds={itemIds} onSelect={onSelect} />
+      </div>
+    </Overflow>
+  );
+};
+
+const OverflowSelectionItem: React.FC<{
+  onSelectItem?: (item: string) => void;
+  selected?: boolean;
+  id: string;
+}> = props => {
+  const onClick = () => {
+    props.onSelectItem?.(props.id);
+  };
+
+  return (
+    <OverflowItem id={props.id} priority={props.selected ? 1000 : undefined}>
+      <Button appearance={props.selected ? 'primary' : 'secondary'} onClick={onClick}>
+        Item {props.id}
+      </Button>
+    </OverflowItem>
+  );
+};
+
+const OverflowMenuItem: React.FC<Pick<OverflowItemProps, 'id'> & { onClick: () => void }> = props => {
+  const { id, onClick } = props;
+  const isVisible = useIsOverflowItemVisible(id);
+
+  if (isVisible) {
+    return null;
+  }
+
+  return <MenuItem onClick={onClick}>Item {id}</MenuItem>;
+};
+
+const OverflowMenu: React.FC<{
+  itemIds: string[];
+  onSelect: (itemId: string) => void;
+}> = ({ itemIds, onSelect }) => {
+  const { ref, overflowCount, isOverflowing } = useOverflowMenu<HTMLButtonElement>();
+
+  if (!isOverflowing) {
+    return null;
+  }
+
+  return (
+    <Menu>
+      <MenuTrigger disableButtonEnhancement>
+        <MenuButton ref={ref}>+{overflowCount} items</MenuButton>
+      </MenuTrigger>
+
+      <MenuPopover>
+        <MenuList>
           {itemIds.map(i => (
-            <TestOverflowItem key={i} id={i} />
+            <OverflowMenuItem onClick={() => onSelect(i)} key={i} id={i} />
           ))}
-          <OverflowMenu itemIds={itemIds} />
-        </div>
-      </Overflow>
-
-      <div className={styles.farItems}>
-        <Button>Foo</Button>
-        <Button>Bar</Button>
-      </div>
-    </div>
+        </MenuList>
+      </MenuPopover>
+    </Menu>
   );
 };
+
+Pinned.parameters = {
+  docs: {
+    description: {
+      story: [
+        'An item can be pinned (always visible) by setting it to be a higher priority that all other overflow items.',
```

Comment:
The pinned item is not always visible in the demo.


https://user-images.githubusercontent.com/93940821/210869672-abb2082f-9ccb-4214-a755-318d4ef72d0c.mp4

Reply:
This is an expected edge case, when there is no space for _**any item**_, it should be normal that the pinned item is finally overflowed. I'll add some extra docs to that effect

Reply:
Added extra docs in https://github.com/microsoft/fluentui/pull/26183/commits/6a8450795049a7baefd0fe4a871ae6b0bc079d84